### ISSUE TITLE
As a bot developer I want to have more control on the entities of my dialog when reentrant

### ISSUE THREAD

**Issue by yangeorget** [Botfuel/botfuel-dialog#234]
If I have an intent (**research-operation**) with two entities:
- **category**
- **subcategory**

**subcategories are associated to a specific category**, but at some points when we reenter in the dialog with new **subcategory** or **category** this new entity don't match this associated subcategory/category and Emmanuel have too update the dialog entities manually in his dialog to make sure that the new entities are associated to the previous matched.

It would be nice to have a new entity hook to be able to update matched entities values so that in this case Emmanuel would be able to reset the previous matched entities if they don't match with the new ones.

**Comment by yangeorget** [comment]
➤ Sander commented:

Original request:

https://botfuel.productboard.com/inbox/notes/2248019

**Comment by yangeorget** [comment]
➤ Sander commented:

[Yan Georget](https://app.asana.com/0/110937017121846/list)[Emmanuel Steinitz](https://app.asana.com/0/695807215959766/list) This needs to be dicussed further between the two of you before we can proceed.

**Comment by botfuel-unitobot** [comment]
➤ Maxime Khoy commented:

[Emmanuel Steinitz](https://app.asana.com/0/695807215959766/list) two ways of doing this will be documented by this: [Explain use of entities in Prompt dialog](https://app.asana.com/0/370953015396593/802384541686725)
